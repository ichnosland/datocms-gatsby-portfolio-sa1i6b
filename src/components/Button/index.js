/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
