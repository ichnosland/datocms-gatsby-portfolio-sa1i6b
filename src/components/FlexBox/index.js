/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
