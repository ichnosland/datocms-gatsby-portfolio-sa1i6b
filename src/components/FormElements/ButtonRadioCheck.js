/**
*
* ButtonRadioCheck
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { colore } from 'style/color';
import { RadioCheckInput } from './InputRadioCheck';

export const ButtonRadioCheckLabel = styled.label`
  position: relative;
  display: block;
  padding: 12px;
  margin: 10px 0;
  background-color: ${colore.ui.contrast};
  border: 1px solid ${(props) => props.theme.brand};
  border-bottom-width: 3px;
  border-radius: ${(props) => props.theme.radius.general};
  ${({ checked }) => checked && css`
    color: ${(props) => props.theme.brand};
    border-width: 1px;
    &:after{
      content: '';
      display: block;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      border-radius: 5px;
      border: 3px solid ${(props) => props.theme.pale};
    }
  `};
`;

ButtonRadioCheckLabel.defaultProps = {
  theme: {
    brand: '#00BBEF',
    pale: 'rgb(199, 231, 245)',
    radius: {
      general: '6px',
    },
  },
};

class ButtonRadioCheck extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { label, type, id, input } = this.props;
    return (
      <ButtonRadioCheckLabel
        type={type}
        checked={input.checked}
        htmlFor={id}
      >
        <RadioCheckInput
          {...input}
          type={type}
          id={id}
        />
        {label}
      </ButtonRadioCheckLabel>
    );
  }
}

ButtonRadioCheck.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  id: PropTypes.string,
  input: PropTypes.object,
};

export default ButtonRadioCheck;
