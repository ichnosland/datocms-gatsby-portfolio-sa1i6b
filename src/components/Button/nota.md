Poich√© i Button sono uno styled-components le icone sono a loro volta uno s-c a parte, **`<Icon />`**, da inserire manualmente come *children*.
Anche le icone per i pulsanti sono dei path a parte in 'icons/buttons.js'.

Quindi:
  ~~~js
  import { Button, Icon } from '../Button';
  import buttonicon from 'icons/buttons';

  <!-- ... -->

  <Button>
    <Icon {...buttonicon.nomeIcona} />
  </Button>
  ~~~