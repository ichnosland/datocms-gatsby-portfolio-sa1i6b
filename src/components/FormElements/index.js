/**
*
* FormElements
*
*/

import styled, { css } from 'styled-components';

import Section from 'components/Section';
import Div from 'components/Div';
import { Button, Icon } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const BasicInputStyle = css`
  display: block;
  box-sizing: border-box;
  color: ${(props) => props.theme.brand};
  width: 100%;
  text-align: ${(props) => props.align ? props.align : 'left'};
  appearance: none;
  padding: 6px 13px;
  border: 1px solid ${(props) => props.theme.light};
  background-color: ${colore.ui.contrast};
  ${({ note }) => note && css`
    color: ${colore.ui.txt};
    background-color: ${colore.ui.note};
    border-color: ${colore.ui.noteBorder};
  `}
  &::placeholder {
    position: relative;
    color: ${(props) => props.theme.subtle};
    ${({ note }) => note && css`
      color: ${colore.ui.noteBorder};
    `}
    opacity: 1;
    overflow: visible;
  }
  &:active,
  &:focus {
    outline: none;
  }
`;

BasicInputStyle.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

export const FormSection = styled(Section)`
  height: 100%;
  flex-direction: column;
  justify-content: space-between;
  &::before {
    content: "";
    display: inline;
  }
`;

export const CardForm = styled.form`
  width: 280px;
  margin: 0 auto;
  text-align: center;
  ${media.lt480`
    width: 240px;
  `}
`;

export const FormElement = styled(Div)`
  position: relative;
  margin: ${(props) => props.margin ? props.margin : '0 auto 18px'};
  ${(props) => props.errore && css`
    margin-bottom: 9px;
  }`
}`;


export const FormMessage = styled.span`
  display: inline-block;
  width: 100%;
  font-size: 15px;
  color: ${(props) => props.color ? props.color : colore.actions.error};
  padding: 6px 9px;
  margin-top: 9px;
  border: 1px solid ${colore.ui.neutralBg};
  background-color: ${colore.ui.contrast};
  border-radius: ${(props) => props.theme.radius.general};
  ${(props) => props.landing && css`
    font-size: smaller;
    padding: 2px 12px;
    border-radius: 60px;
    margin-bottom: 3px;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
`;

FormMessage.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export const FacebookButton = styled(Button)`
  background-color: ${colore.social.facebook};
  border-radius: 3px;
`;

export const FacebookIcon = styled(Icon)`
  top: -2px;
  fill: #fff;
`;
