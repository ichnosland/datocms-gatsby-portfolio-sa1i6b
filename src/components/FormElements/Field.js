/**
*
* Field
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { colore } from 'style/color';
import FlexBox from '../FlexBox';
import Spinner from '../Spinner';
import { BasicInputStyle, FormMessage } from './index';

export const Input = styled.input`
  ${BasicInputStyle};
  &[type="search"] {
    box-sizing: border-box;
    text-align: left;
    border-radius: ${(props) => props.theme.radius.searchBox};
    background-color: ${colore.ui.contrast};
  }
  ${(props) => props.landing && css`
    order: 2;
    margin-bottom: 12px;
    width: 100%;
    appearance: none;
    background-color: white;
    background-image: none;
    padding: 6px 13px 6px 13px;
    border: none;
    border-radius: 3px;
    &::placeholder {
      position: relative;
      color: rgba(0, 0, 0, 0.3);
      opacity:1;
      overflow: visible;
    }
  `}
`;

Input.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

class Field extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { input, note, align, label, type, meta: { asyncValidating, touched, error }, landing } = this.props;
    return (
      <FlexBox direction="column">
        <Input
          {...input}
          placeholder={label}
          type={type}
          note={note}
          align={align}
          landing={landing}
        />
        {asyncValidating && <Spinner size="small" />}
        {touched && error && <FormMessage landing={landing}>{error}</FormMessage>}
      </FlexBox>
    );
  }
}

Input.defaultProps = {
  theme: {
    radius: {
      searchBox: '100px',
    },
  },
};

Field.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  note: PropTypes.bool,
  landing: PropTypes.bool,
  align: PropTypes.string,
};

export default Field;
