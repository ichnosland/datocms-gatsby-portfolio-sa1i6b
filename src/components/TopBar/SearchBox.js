/**
*
* SearchBox
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import FlexBox from 'components/FlexBox';
import { Input } from 'components/FormElements/Field';
import { Button } from 'components/Button';
import Svg from 'components/Svg';
import { colore } from 'style/color';
import icon from 'icons/buttons';
import globalicon from 'icons/globals';
import media from 'style/mediainjector';

export const SearchBoxWrap = styled(FlexBox)`
  flex-grow: 1;
  min-width: 80px;
  height: 56px;
  transition: all 0.5s;
  ${({ searchActive }) => searchActive && css`
    z-index: 100;
    width: 100%;
    margin-left: 10px;
  `};
`;

const searchBoxIconStyle = css`
  height: 100%;
  position: absolute;
  top: 0;
  svg {
    display: block;
    width: 18px;
    height: 18px;
  }
`;

export const Lens = styled(FlexBox)`
  ${searchBoxIconStyle}
  width: 36px;
  left:0;
  svg {
    fill: ${colore.ui.darkBg};
  }
`;

export const CancelBtn = styled(Button)`
  ${searchBoxIconStyle}
  width: 36px;
  right:0;
  margin: 0;
  padding: 0;
  background: none;
  visibility: hidden;
  ${media.iphone4`
    padding: 0;
    svg {
      margin: 0;
    }
  `}
  ${({ searchActive }) => searchActive && css`
    visibility: visible;
  `};
`;

export const SearchField = styled(Input)`
  color: ${colore.ui.txt};
  padding-left: 32px;
  padding-right: 26px;
  margin: 0;
  border: none;
  font-size: 18px;
  &::placeholder {
    color: ${colore.ui.darkBg};
  }
  &:active,
  &:focus {
    outline: none;
  }
`;

export const PremiumIcon = styled(FlexBox)`
  width: 32px;
  height: 32px;
  margin: 2px 2px 2px 12px;
  background-color: ${colore.actions.action};
  border-radius: 300px;
  svg {
    fill: ${colore.ui.contrast};
  }
`;

export default class SearchBox extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      value,
      className,
      placeholder,
      onFiltraUnita,
      searchActive,
      onActivateSearch,
      searchResetFunction,
      premiumActive,
    } = this.props;

    return (
      <SearchBoxWrap searchActive={searchActive}>
        <Lens>
          <Svg {...icon.lens} />
        </Lens>
        <SearchField
          type="search"
          value={value}
          className={className}
          placeholder={placeholder}
          onChange={(e) => onFiltraUnita(e.target.value)}
          onFocus={onActivateSearch}
        />
        {searchResetFunction && (
          <CancelBtn
            searchActive={searchActive && value}
            onClick={searchResetFunction}
          >
            <Svg {...icon.cancel} />
          </CancelBtn>)}
        {premiumActive &&
          <PremiumIcon>
            <Svg {...globalicon.starCircle} />
          </PremiumIcon>
        }
      </SearchBoxWrap>
    );
  }
}

SearchBox.propTypes = {
  onFiltraUnita: PropTypes.func.isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  searchActive: PropTypes.bool.isRequired,
  onActivateSearch: PropTypes.func.isRequired,
  searchResetFunction: PropTypes.func,
  premiumActive: PropTypes.bool,
};
