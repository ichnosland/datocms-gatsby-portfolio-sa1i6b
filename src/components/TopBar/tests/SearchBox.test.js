import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '../Button';
import FlexBox from '../FlexBox';
import { Input } from '../FormElements/Field';
import SearchBox, { SearchBoxWrap, SearchField, Lens, CancelBtn, PremiumIcon } from '../SearchBox';

const type = 'text';
const searchActive = false;

describe('<SearchBox />', () => {
  it('should render a SearchBoxWrap', () => {
    const renderedComponent = shallow(<SearchBox
      onFiltraUnita={() => { }}
      searchActive={searchActive}
      onActivateSearch={() => { }}
    />);
    expect(renderedComponent.find(SearchBoxWrap).length).toEqual(1);
  });
  it('should render its css /w props searchActive', () => {
    const renderedComponent = shallow(<SearchBox
      onFiltraUnita={() => { }}
      searchActive
      onActivateSearch={() => { }}
    />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render its props value', () => {
    const value = 'test';
    const renderedComponent = shallow(<SearchBox
      value={value}
      onFiltraUnita={() => { }}
      searchActive={searchActive}
      onActivateSearch={() => { }}
    />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should call onFiltraUnita when onChange is called', () => {
    const value = 'test';
    const props = {
      value,
      onFiltraUnita: jest.fn(),
      searchActive: false,
      onActivateSearch: () => { },
    };
    const renderedComponent = shallow(<SearchBox
      {...props}
    />);
    renderedComponent.find(SearchField).simulate('change', { target: { value: 'some value' } });
    expect(props.onFiltraUnita).toHaveBeenCalled();
  });

  it('should call onActivateSearch when onFocus is called', () => {
    const props = {
      value: '',
      onFiltraUnita: () => {},
      onActivateSearch: jest.fn(),
      searchActive: false,
    };
    const renderedComponent = shallow(<SearchBox
      {...props}
    />);
    renderedComponent.find(SearchField).simulate('focus');
    expect(props.onActivateSearch).toHaveBeenCalled();
  });

  it('should call onActivateSearch when onClick is called', () => {
    const props = {
      value: 'findme',
      onFiltraUnita: () => {},
      onActivateSearch: () => {},
      searchResetFunction: jest.fn(),
      searchActive: false,
    };
    const renderedComponent = shallow(<SearchBox
      {...props}
    />);
    renderedComponent.find(CancelBtn).simulate('click');
    expect(props.searchResetFunction).toHaveBeenCalled();
  });

  it('should render its css /w props searchResetFunction & searchActive & value', () => {
    const renderedComponent = shallow(
      <SearchBox
        onActivateSearch={() => {}}
        searchResetFunction={() => { }}
        searchActive
        value="ciao"
        onFiltraUnita={() => { }}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
    expect(renderedComponent.find(CancelBtn).length).toEqual(1);
  });

  it('should render its css /w props premiumActive', () => {
    const renderedComponent = shallow(<SearchBox
      onFiltraUnita={() => { }}
      premiumActive
      searchActive={false}
      onActivateSearch={() => { }}
    />);
    expect(renderedComponent.find(PremiumIcon).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SearchBoxWrap />', () => {
  it('should render a FlexBox', () => {
    const renderedComponent = shallow(<SearchBoxWrap />);
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
  });
  it('should render its css', () => {
    const renderedComponent = shallow(<SearchBoxWrap />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props searchActive true', () => {
    const renderedComponent = shallow(<SearchBoxWrap searchActive />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SearchField />', () => {
  it('should render a Input', () => {
    const renderedComponent = shallow(<SearchField type={type} />);
    expect(renderedComponent.find(Input).length).toEqual(1);
  });

  it('should render its css', () => {
    const renderedComponent = shallow(<SearchField type={type} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Lens />', () => {
  it('should render a FlexBox', () => {
    const renderedComponent = shallow(<Lens />);
    expect(renderedComponent.find(FlexBox).length).toEqual(1);
  });
  it('should render its css', () => {
    const renderedComponent = shallow(<Lens />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<CancelBtn />', () => {
  it('should render a Button', () => {
    const renderedComponent = shallow(<CancelBtn />);
    expect(renderedComponent.find(Button).length).toEqual(1);
  });
  it('should render its css', () => {
    const renderedComponent = shallow(<CancelBtn />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css /w props searchActive', () => {
    const renderedComponent = shallow(<CancelBtn searchActive />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
