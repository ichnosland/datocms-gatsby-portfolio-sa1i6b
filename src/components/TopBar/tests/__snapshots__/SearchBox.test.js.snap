// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CancelBtn /> should render its css /w props searchActive 1`] = `
.c0 {
  height: 100%;
  position: absolute;
  top: 0;
  width: 36px;
  right: 0;
  margin: 0;
  padding: 0;
  background: none;
  visibility: hidden;
  visibility: visible;
}

.c0 svg {
  display: block;
  width: 18px;
  height: 18px;
}

@media (min-device-width:20em) and (max-device-height:30em) {
  .c0 {
    padding: 0;
  }

  .c0 svg {
    margin: 0;
  }
}

<Button
  actioncolor=""
  className="c0"
  searchActive={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "fonts": Object {
        "buttons": Object {
          "fontFamily": "inherit",
          "fontWeight": "inherit",
        },
      },
      "radius": Object {
        "buttons": "6px",
      },
    }
  }
/>
`;

exports[`<CancelBtn /> should render its css 1`] = `
.c0 {
  height: 100%;
  position: absolute;
  top: 0;
  width: 36px;
  right: 0;
  margin: 0;
  padding: 0;
  background: none;
  visibility: hidden;
}

.c0 svg {
  display: block;
  width: 18px;
  height: 18px;
}

@media (min-device-width:20em) and (max-device-height:30em) {
  .c0 {
    padding: 0;
  }

  .c0 svg {
    margin: 0;
  }
}

<Button
  actioncolor=""
  className="c0"
  theme={
    Object {
      "brand": "#00BBEF",
      "fonts": Object {
        "buttons": Object {
          "fontFamily": "inherit",
          "fontWeight": "inherit",
        },
      },
      "radius": Object {
        "buttons": "6px",
      },
    }
  }
/>
`;

exports[`<Lens /> should render its css 1`] = `
.c0 {
  height: 100%;
  position: absolute;
  top: 0;
  width: 36px;
  left: 0;
}

.c0 svg {
  display: block;
  width: 18px;
  height: 18px;
}

.c0 svg {
  fill: rgb(209,209,209);
}

<FlexBox
  className="c0"
/>
`;

exports[`<SearchBox /> should render its css /w props premiumActive 1`] = `
<SearchBox__SearchBoxWrap
  searchActive={false}
>
  <SearchBox__Lens>
    <Svg
      height="18px"
      path={
        <path
          d="M7.17 15.624c0-4.665 3.797-8.454 8.458-8.454 4.66 0 8.456 3.795 8.456 8.454 0 4.66-3.796 8.454-8.456 8.454-4.661 0-8.457-3.795-8.457-8.454zm17.628 5.338a10.553 10.553 0 0 0 1.457-5.338C26.255 9.764 21.49 5 15.628 5 9.765 5 5 9.764 5 15.624s4.765 10.624 10.628 10.624c1.95 0 3.766-.534 5.34-1.45l5.409 5.407a2.709 2.709 0 0 0 3.83 0 2.717 2.717 0 0 0 0-3.835l-5.409-5.408z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 36 36"
      width="18px"
    />
  </SearchBox__Lens>
  <SearchBox__SearchField
    onChange={[Function]}
    onFocus={[Function]}
    type="search"
  />
  <SearchBox__PremiumIcon>
    <Svg
      height="32px"
      path={
        <path
          d="M16 0C7.164 0 0 7.164 0 16s7.164 16 16 16 16-7.164 16-16S24.836 0 16 0zm10.98 12.21c-.103-.215-.31-.376-.515-.376h-7.423l-2.527-7.458C16.412 4.16 16.205 4 16 4c-.207 0-.413.16-.516.376l-2.526 7.458H5.534c-.206 0-.412.16-.515.376-.053.215 0 .483.205.59L11 17.63l-2.58 7.618c-.102.215 0 .483.156.59.155.162.413.215.62.055L16 21.653l6.804 4.24c.103.054.207.107.26.107.154 0 .257-.054.36-.16.155-.11.258-.377.155-.592L21 17.628l5.775-4.827c.206-.107.258-.375.206-.59z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 32 32"
      width="32px"
    />
  </SearchBox__PremiumIcon>
</SearchBox__SearchBoxWrap>
`;

exports[`<SearchBox /> should render its css /w props searchActive 1`] = `
<SearchBox__SearchBoxWrap
  searchActive={true}
>
  <SearchBox__Lens>
    <Svg
      height="18px"
      path={
        <path
          d="M7.17 15.624c0-4.665 3.797-8.454 8.458-8.454 4.66 0 8.456 3.795 8.456 8.454 0 4.66-3.796 8.454-8.456 8.454-4.661 0-8.457-3.795-8.457-8.454zm17.628 5.338a10.553 10.553 0 0 0 1.457-5.338C26.255 9.764 21.49 5 15.628 5 9.765 5 5 9.764 5 15.624s4.765 10.624 10.628 10.624c1.95 0 3.766-.534 5.34-1.45l5.409 5.407a2.709 2.709 0 0 0 3.83 0 2.717 2.717 0 0 0 0-3.835l-5.409-5.408z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 36 36"
      width="18px"
    />
  </SearchBox__Lens>
  <SearchBox__SearchField
    onChange={[Function]}
    onFocus={[Function]}
    type="search"
  />
</SearchBox__SearchBoxWrap>
`;

exports[`<SearchBox /> should render its css /w props searchResetFunction & searchActive & value 1`] = `
<SearchBox__SearchBoxWrap
  searchActive={true}
>
  <SearchBox__Lens>
    <Svg
      height="18px"
      path={
        <path
          d="M7.17 15.624c0-4.665 3.797-8.454 8.458-8.454 4.66 0 8.456 3.795 8.456 8.454 0 4.66-3.796 8.454-8.456 8.454-4.661 0-8.457-3.795-8.457-8.454zm17.628 5.338a10.553 10.553 0 0 0 1.457-5.338C26.255 9.764 21.49 5 15.628 5 9.765 5 5 9.764 5 15.624s4.765 10.624 10.628 10.624c1.95 0 3.766-.534 5.34-1.45l5.409 5.407a2.709 2.709 0 0 0 3.83 0 2.717 2.717 0 0 0 0-3.835l-5.409-5.408z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 36 36"
      width="18px"
    />
  </SearchBox__Lens>
  <SearchBox__SearchField
    onChange={[Function]}
    onFocus={[Function]}
    type="search"
    value="ciao"
  />
  <SearchBox__CancelBtn
    onClick={[Function]}
    searchActive="ciao"
  >
    <Svg
      height="18px"
      path={
        <path
          d="M18 34C9.163 34 2 26.837 2 18S9.163 2 18 2s16 7.163 16 16-7.163 16-16 16zm3-16l6-6-3-3-6 6-6-6-3 3 6 6-6 6 3 3 6-6 6 6 3-3-6-6z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 36 36"
      width="18px"
    />
  </SearchBox__CancelBtn>
</SearchBox__SearchBoxWrap>
`;

exports[`<SearchBox /> should render its props value 1`] = `
<SearchBox__SearchBoxWrap
  searchActive={false}
>
  <SearchBox__Lens>
    <Svg
      height="18px"
      path={
        <path
          d="M7.17 15.624c0-4.665 3.797-8.454 8.458-8.454 4.66 0 8.456 3.795 8.456 8.454 0 4.66-3.796 8.454-8.456 8.454-4.661 0-8.457-3.795-8.457-8.454zm17.628 5.338a10.553 10.553 0 0 0 1.457-5.338C26.255 9.764 21.49 5 15.628 5 9.765 5 5 9.764 5 15.624s4.765 10.624 10.628 10.624c1.95 0 3.766-.534 5.34-1.45l5.409 5.407a2.709 2.709 0 0 0 3.83 0 2.717 2.717 0 0 0 0-3.835l-5.409-5.408z"
          fillRule="evenodd"
        />
      }
      viewBox="0 0 36 36"
      width="18px"
    />
  </SearchBox__Lens>
  <SearchBox__SearchField
    onChange={[Function]}
    onFocus={[Function]}
    type="search"
    value="test"
  />
</SearchBox__SearchBoxWrap>
`;

exports[`<SearchBoxWrap /> should render its css /w props searchActive true 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 80px;
  height: 56px;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  z-index: 100;
  width: 100%;
  margin-left: 10px;
}

<FlexBox
  className="c0"
  searchActive={true}
/>
`;

exports[`<SearchBoxWrap /> should render its css 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 80px;
  height: 56px;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<FlexBox
  className="c0"
/>
`;

exports[`<SearchField /> should render its css 1`] = `
.c0 {
  color: rgb(78,78,78);
  padding-left: 32px;
  padding-right: 26px;
  margin: 0;
  border: none;
  font-size: 18px;
}

.c0::-webkit-input-placeholder {
  color: rgb(209,209,209);
}

.c0::-moz-placeholder {
  color: rgb(209,209,209);
}

.c0:-ms-input-placeholder {
  color: rgb(209,209,209);
}

.c0::placeholder {
  color: rgb(209,209,209);
}

.c0:active,
.c0:focus {
  outline: none;
}

<Field__Input
  className="c0"
  theme={
    Object {
      "radius": Object {
        "searchBox": "100px",
      },
    }
  }
  type="text"
/>
`;
